src/module/rules/rule-element/token-light.ts

```diff
@@ -14,60 +14,59 @@ class TokenLightRuleElement extends RuleElementPF2e<TokenLightRuleSchema> {
             ...super.defineSchema(),
             value: new fields.SchemaField({
                 ...foundry.data.LightData.defineSchema(),
-                dim: new ResolvableValueField({ required: false, nullable: false, initial: undefined }),
                 bright: new ResolvableValueField({ required: false, nullable: false, initial: undefined }),
+                color: new fields.StringField({ required: false, nullable: true, blank: false, initial: null }),
+                dim: new ResolvableValueField({ required: false, nullable: false, initial: undefined }),
             }),
         };
     }
 
-    constructor(data: RuleElementSource, options: RuleElementOptions) {
-        super(data, options);
-        this.validateData();
-    }
-
-    validateData(): void {
+    getLightData(): SourceFromSchema<LightDataSchema> | null {
         const light = this.value;
 
-        for (const key of ["dim", "bright"] as const) {
+        light.color &&= this.resolveInjectedProperties(light.color);
+
```

Comment:
Would it be worth validating the resolved color through isColorString()?

Reply:
already done